src/App.tsx

```diff
@@ -16,24 +23,46 @@ import 'normalize.css'
 import './App.css'
 
 let checkCardInterval = 0
+let checkStatusInterval = 0
 
 let loadingElection = false
 
 const App: React.FC = () => {
-  const [isProgrammingCard, setIsProgrammingCard] = useState(false)
   const [election, setElection] = useStateAndLocalStorage<OptionalElection>(
     'election'
   )
+  const [status, setStatus] = useState<ScannerStatus>({ batches: [] })
+  const { batches } = status
+  const isScanning = batches && batches[0] && !batches[0].endedAt
```

Comment:
I'll have to make sure I'm returning batches in the right order here! I'll check on that. Your code is good though.